### PR TITLE
Enable actor methods to be decorated on the caller side also and get postprocessors.

### PR DESCRIPTION
**Pull request by robertnishihara**
When we decorate an actor method, the actor will execute the decorated method. However, the caller will call the undecorated method. It'd be helpful to have a way to run decorated methods on both the caller and callee side. This is one way to do it.

This is very much a quick pass at it, and doesn't cover a lot of cases.

TODO
- [x] add tests

These are the changes needed to enable #4501 cleanly.

cc @visatish

### CODE REVIEW

File: python/ray/actor.py [hunk 1]
```diff
@@ -109,10 +109,16 @@ def annotate_method(method):
 # Create objects to wrap method invocations. This is done so that we can
 # invoke methods with actor.method.remote() instead of actor.method().
 class ActorMethod(object):
-    def __init__(self, actor, method_name, num_return_vals):
+    def __init__(self, actor, method_name, num_return_vals, decorator=None):
         self._actor = actor
         self._method_name = method_name
         self._num_return_vals = num_return_vals
+        # This is a decorator that is used to wrap the function invocation (as
```
Comment (pcmoritz) [inline python/ray/actor.py]: This should be in the docstring of the __init__ method, and also point the user to an example on how to use it.

File: python/ray/actor.py [hunk 2]
```diff
@@ -196,6 +209,7 @@ def __init__(self):
         # Extract the signatures of each of the methods. This will be used
         # to catch some errors if the methods are called with inappropriate
         # arguments.
+        self._method_decorators = {}
```
Comment (pcmoritz) [inline python/ray/actor.py]: Add documentation to the ActorClass Attributes section

**Comment by pcmoritz** [comment]
This looks good. We might also want to make it work for tasks.

**Comment by robertnishihara** [comment]
Thanks @pcmoritz! I fixed everything and made it work for remote functions.

**Review by pcmoritz — APPROVED** [review]

**Comment by guoyuhong** [comment]
@robertnishihara Just curious. Current solution only supports one decorator for a function right? Maybe later, multiple decorators will be supported, right?

**Comment by robertnishihara** [comment]
@guoyuhong that's a good point. I don't like the API currently. Maybe a better way to do it would be to allow a `caller_decorators=[...]` argument to the `ray.remote` decorator for remote functions and the `ray.method` decorator for actor methods.

**Pull request merged**